plugins/repoquery.py

```diff
@@ -293,16 +291,20 @@ def run(self, args):
                     self.base.sack, with_provides=False)
                 pkgs += q.run()
             q = self.base.sack.query().filter(pkg=pkgs)
-        else:
-            q = self.base.sack.query()
+
+        if self.opts.list == "recent":
+            q.recent(self.base.conf.recent)
+        elif self.opts.list == "autoremove":
+            q.autoremove(self.base.sack, self.base.yumdb)
+        elif self.opts.list:
+            q = getattr(q, self.opts.list)()
 
         if self.opts.pkgfilter == "duplicated":
-            dups = dnf.query.duplicated_pkgs(q, self.base.conf.installonlypkgs)
-            q = q.filter(pkg=dups)
+            installonly = self.installonly(q)
+            exclude = [pkg.name for pkg in installonly]
+            q = q.duplicated(exclude)
```

Comment:
your commit actually fixes [1], reference this in the message, please
[1] https://bugzilla.redhat.com/show_bug.cgi?id=1231572